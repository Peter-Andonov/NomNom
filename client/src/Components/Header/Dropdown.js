import React from 'react';
import styled from 'styled-components';
import StyledLink from './Link';

export default function DropdownMenu() {

    const Container = styled.div`
    position: absolute;
    top: 100px;
    width: auto;
    min-width: 175px;
    display: flex;
    flex-direction: column;
    align-items: center;
    background: #333333;
    border: 1px solid #474a4d;
    border-radius: 5px;
    `

    return (
        <Container>
            <StyledLink to={'/admin'} text={'Admin'} />
            <StyledLink to={'/profile'} text={'Profile'} />
            <StyledLink to={'/logout'} text={'Logout'} />
        </Container>
    );
}