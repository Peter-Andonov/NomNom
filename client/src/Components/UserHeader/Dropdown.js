import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import styled from 'styled-components';
import UserContext from '../../Context';
import StyledLink from './Link';


const Container = styled.div`
    position: absolute;
    top: 100px;
    width: auto;
    min-width: 175px;
    display: flex;
    flex-direction: column;
    align-items: center;
    background: #333333;
    border: 1px solid #474a4d;
    border-radius: 5px;
`;

const Div = styled.div`
    margin: 0.5rem;
    color: white;
    size: 0.8rem;
    &:hover {
        color: rgba(237, 71, 59);
        cursor: pointer;
    }
`

class DropdownMenu extends Component {

    static contextType = UserContext;

    logOut = (e) => {
        this.context.logOut();
        this.props.history.push('/');
    }

    render() {
        return (
            <Container>
                {this.context.user.role === 'admin' && <StyledLink to={'/admin'} text={'Admin'} />}
                {this.context.user.role === 'user' && <StyledLink to={'/favourites'} text={'Favourites'} />}
                <StyledLink to={`/profile/${this.context.user._id}`} text={'Profile'} />
                <Div onClick={this.logOut}>Logout</Div>
            </Container>
        );
    };
};

export default withRouter(DropdownMenu);