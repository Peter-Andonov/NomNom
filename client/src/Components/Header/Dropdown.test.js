import React from 'react'
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TestSetup from '../../Utils/TestUtils/test-utils';


import Dropdown from './ProfileNav.js';

describe('Dropdown component', () => {
    it("renders user options - favourites, profile, logout", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: true,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'user',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: undefined,
                    favouriteRecipes: []
                }
            }}>
                <Dropdown />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders admin options - profile, logout", () => {
        const { asFragment } = render(
            <TestSetup value={{
                loggedIn: false,
                user: {
                    _id: '1234',
                    email: "john.doe@gmail.com",
                    role: 'admin',
                    firstName: undefined,
                    lastName: undefined,
                    profilePicUrl: '',
                    favouriteRecipes: []
                }
            }}>
                <Dropdown />
            </TestSetup>);
        expect(asFragment()).toMatchSnapshot();
    });
});