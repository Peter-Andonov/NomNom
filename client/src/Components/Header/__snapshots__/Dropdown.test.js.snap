// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component renders admin options - profile, logout 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hVzgMg"
  >
    <img
      alt="Avatar"
      class="sc-AxiKw fiBQpN"
      src="https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png"
    />
    <div
      class="sc-AxjAm cxNAeo"
    >
      john.doe@gmail.com
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dropdown component renders user options - favourites, profile, logout 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ hVzgMg"
  >
    <img
      alt="Avatar"
      class="sc-AxiKw fiBQpN"
      src="https://res.cloudinary.com/nomnomapp/image/upload/v1596641498/Images/x5svb4l6kwcyiinazfqt.png"
    />
    <div
      class="sc-AxjAm cxNAeo"
    >
      john.doe@gmail.com
    </div>
  </div>
</DocumentFragment>
`;
